how to compile it in linux

```
# g++ sfpack.cpp
sfpack.cpp:60:83: error: ‘function’ in namespace ‘std’ does not name a type
 void ReadDirectory(const std::string& dirPath, FILE* f, uint64_t dirOffset, const std::function<const char*(uint64_t)>& getName)
                                                                                   ^
sfpack.cpp:60:88: error: ISO C++ forbids declaration of ‘parameter’ with no type [-fpermissive]
 void ReadDirectory(const std::string& dirPath, FILE* f, uint64_t dirOffset, const std::function<const char*(uint64_t)>& getName)
                                                                                        ^
sfpack.cpp:60:96: error: expected ‘,’ or ‘...’ before ‘<’ token
 void ReadDirectory(const std::string& dirPath, FILE* f, uint64_t dirOffset, const std::function<const char*(uint64_t)>& getName)
                                                                                                ^
sfpack.cpp: In function ‘void ReadDirectory(const string&, FILE*, uint64_t, int)’:
sfpack.cpp:67:52: error: ‘getName’ was not declared in this scope
  std::string fn = (dirPath + getName(dir.nameOffset));
                                                    ^
sfpack.cpp: In function ‘int main(int, char**)’:
sfpack.cpp:155:2: warning: lambda expressions only available with -std=c++11 or -std=gnu++11 [enabled by default]
  });
  ^
sfpack.cpp:155:3: error: cannot convert ‘main(int, char**)::__lambda0’ to ‘int’ for argument ‘4’ to ‘void ReadDirectory(const string&, FILE*, uint64_t, int)’
  });
   ^
```

--- a/Makefile
+++ b/Makefile
@@ -1,0 +1,19 @@
+# the compiler: gcc for C program, define as g++ for C++
+CC = g++
+
+# compiler flags:
+#  -g    adds debugging information to the executable file
+#  -Wall turns on most, but not all, compiler warnings
+# std=c++11 author req on https://github.com/nta/sfpack/issues/2
+CFLAGS  = -g -Wall -std=c++11
+
+# the build target executable:
+TARGET = sfpack
+
+all: $(TARGET)
+
+$(TARGET): $(TARGET).cpp
+	$(CC) $(CFLAGS) -o $(TARGET) $(TARGET).cpp
+
+clean:
+	$(RM) $(TARGET)

--- a/README.md
+++ b/README.md
@@ -10,6 +10,11 @@ sfpack Y:\dls\system.common.sfp
 
 have fun.
 
+# Building
+
+Just run `make`, this requires `g++` (provided by `build-essential` on Ubuntu).
+You can install `build-essential` with `sudo apt-get install build-essential`.
+
 ## licensing-ish
 
 ```

